### DISCUSSION TITLE
Content-Type for JSONResponse defaults to text/html

### DISCUSSION THREAD

**Discussion by bburtin** [fastapi/fastapi#11422] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.get("/v2/properties/{clip}/buildings", response_class=JSONResponse)
def get_buildings(request: Request, clip: int) -> Dict[str, Any]:
    return _templates.TemplateResponse(
        request,
        "buildings.json",
        context={"clip": clip},
        # media_type="application/json",
    )
```

### Description

When I return a `JSONResponse` based on a Jinja template file stored on disk, FastAPI sets the **Content-Type** header to `text/html`.  If I set `media_type="application/json"`, it returns `application/json`.  Is this working as expected?  I expected FastAPI to return `Content-Type: application/json` by default.

This is what I see with the HTTPie command line client:

```text
$ http -v get localhost:8000/v2/properties/123/buildings
GET /v2/properties/123/buildings HTTP/1.1
Accept: */*
Accept-Encoding: gzip, deflate
Connection: keep-alive
Host: localhost:8000
User-Agent: HTTPie/2.6.0

HTTP/1.1 200 OK
content-length: 6856
content-type: text/html; charset=utf-8
date: Wed, 10 Apr 2024 19:06:52 GMT
server: uvicorn

{
}
```

### Operating System

Linux

### FastAPI Version

0.110.0

### Pydantic Version

2.5.3

### Python Version

Python 3.11.1

**Comment by jgould22** [comment, marked as answer]
>  Is this working as expected

Bases on the code in your example it is working as expected

>  I expected FastAPI to return Content-Type: application/json by default.

FastAPI does generally do this.

> When I return a JSONResponse based on a Jinja template file stored on disk

Your code is not doing this, you are returning a [templated response ](https://github.com/encode/starlette/blob/master/starlette/templating.py#L28) which inherits from an [HtmlResponse](https://github.com/encode/starlette/blob/master/starlette/responses.py#L165) which has a content type of `text/html`

It feels like you are misunderstanding how Templates work as this is a bit of an odd way to go about it.

If you want to return json you should define whatever is in 'buildings.json' as a type and return that instead.

**Comment by bburtin** [comment]
Thanks for clearing this up.  I didn't realize that a templated response was meant to be used for HTML.
